File: frontend/src/features/queries/api/queriesController.ts
import api from "../../../shared/api/api";
import { ApiResponse } from "../../../shared/types/ApiResponse";
import { QueryDto } from "../types/QueryDto";
import { CreateQueryDto } from "../types/requests/CreateQueryDto";
import { UpdateQueryDto } from "../types/requests/UpdateQueryDto";

export const queriesController = {
    getQuery: async (
        trackerId: string,
        queryId: string
    ): Promise<ApiResponse<QueryDto>> => {
        return await api.get(`trackers/${trackerId}/queries/${queryId}`);
    },
    getQueryList: async (trackerId: string): Promise<ApiResponse<QueryDto[]>> => {
        return await api.get(`trackers/${trackerId}/queries`);
    },
    createQuery: async (
        trackerId: string,
        values: CreateQueryDto
    ): Promise<ApiResponse<QueryDto>> => {
        return await api.post(`/trackers/${trackerId}/queries`, values);
    },
    updateQuery: async (
        trackerId: string,
        queryId: string,
        values: UpdateQueryDto
    ): Promise<ApiResponse<QueryDto>> => {
        return await api.put(`/trackers/${trackerId}/queries/${queryId}`, values);
    },
    deleteQuery: async (
        trackerId: string,
        queryId: string
    ): Promise<ApiResponse> => {
        return api.delete(`trackers/${trackerId}/queries/${queryId}`);
    },
};

File: backend/src/Operum.API/Controllers/QueriesController.cs
using Microsoft.AspNetCore.Mvc;
using Operum.API.Controllers.Base;
using Operum.Model.DTOs.Queries.Requests;
using Operum.Service.Interfaces;

namespace Operum.API.Controllers
{
    [ApiController]
    [Route("api/trackers/{trackerId}/[controller]")]
    public class QueriesController(IQueriesService queriesService) : BaseController
    {
        [HttpPost]
        public async Task<IActionResult> CreateQuery([FromBody] CreateQueryDto query, [FromRoute] string trackerId)
        {
            return GetApiResponse(await queriesService.CreateQuery(trackerId, query));
        }

        [HttpGet("{queryId}")]
        public async Task<IActionResult> GetQuery([FromRoute] string trackerId, [FromRoute] string queryId)
        {
            return GetApiResponse(await queriesService.GetQuery(trackerId, queryId));
        }

        [HttpGet]
        public async Task<IActionResult> GetQueryList([FromRoute] string trackerId)
        {
            return GetApiResponse(await queriesService.GetQueryList(trackerId));
        }

        [HttpPut("{queryId}")]
        public async Task<IActionResult> UpdateQuery([FromRoute] string trackerId, [FromRoute] string queryId, [FromBody] UpdateQueryDto query)
        {
            return GetApiResponse(await queriesService.UpdateQuery(trackerId, queryId, query));
        }

        [HttpDelete("{queryId}")]
        public async Task<IActionResult> DeleteQuery([FromRoute] string trackerId, [FromRoute] string queryId)
        {
            return GetApiResponse(await queriesService.DeleteQuery(trackerId, queryId));
        }
    }
}
